File: DFS, BFS/7569.py
#토마토2
from collections import deque
def bfs():
    while queue:
        p, q, r = queue.popleft()
        for i in range(6):
            nx = p+dx[i]
            ny = q+dy[i]
            nz = r+dz[i]
            if nx < 0 or nx >= h or ny < 0 or ny >= n or nz < 0 or nz >= m:
                continue
            if graph[nx][ny][nz] == 0:
                graph[nx][ny][nz] = graph[p][q][r] + 1
                queue.append((nx, ny, nz))
    return

graph = []
dx = [-1,1,0,0,0,0]
dy = [0,0,-1,1,0,0]
dz = [0,0,0,0,-1,1]
queue = deque()
result = 0
m, n, h = map(int, input().split())
for _ in range(h):
    temp = []
    for _ in range(n):
        temp.append(list(map(int, input().split())))
    graph.append(temp)

for i in range(h):
    for j in range(n):
        for k in range(m):
            if graph[i][j][k] == 1:
                queue.append((i,j,k))
bfs()
for i in graph:
    for j in i:
        for k in j:
            if k==0:
                print(-1)
                exit(0)
        result = max(result,max(j))
print(result-1)


            

File: DFS, BFS/7569.js
// 토마토
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

let [m, n, h] = input[0].split(" ").map((v) => Number(v));
let graph = [];
let temp = [];
let cnt = 0;
for (let i = 1; i < input.length; i++) {
  temp.push(input[i].split(" ").map((v) => Number(v)));
  if (i % n === 0) {
    graph.push(temp);
    temp = [];
  }
}

let q = [];
for (let i = 0; i < h; i++) {
  for (let j = 0; j < n; j++) {
    for (let k = 0; k < m; k++) {
      if (graph[i][j][k] === 1) q.push([i, j, k]);
    }
  }
}
bfs();
for (let i = 0; i < h; i++) {
  for (let j = 0; j < n; j++) {
    for (let k = 0; k < m; k++) {
      if (graph[i][j][k] === 0) return console.log(-1);
    }
  }
}
console.log(cnt);

function bfs() {
  const dx = [-1, 0, 1, 0, 0, 0];
  const dy = [0, 1, 0, -1, 0, 0];
  const dz = [0, 0, 0, 0, 1, -1];
  let idx = 0;
  while (q.length !== idx) {
    let [x, y, z] = q[idx];
    for (let i = 0; i < 6; i++) {
      let [nx, ny, nz] = [x + dx[i], y + dy[i], z + dz[i]];
      if (
        0 <= nx &&
        nx < h &&
        0 <= ny &&
        ny < n &&
        0 <= nz &&
        nz < m &&
        graph[nx][ny][nz] === 0
      ) {
        cnt = graph[x][y][z];
        graph[nx][ny][nz] = graph[x][y][z] + 1;
        q.push([nx, ny, nz]);
      }
    }
    idx++;
  }
}
